infrastructure/src/main/java/org/wso2/carbon/testgrid/infrastructure/aws/AWSManager.java

```diff
@@ -0,0 +1,186 @@
+        DescribeStacksRequest describeStacksRequest = new DescribeStacksRequest();
+        describeStacksRequest.setStackName(cloudFormationName);
+        DescribeStacksResult describeStacksResult = stackbuilder.describeStacks(describeStacksRequest);
+        //Initially only PublicDNS which is exported from a dummy script is saved in the Deployment object.
+        List<Host> hosts = new ArrayList<>();
+        for (Stack st : describeStacksResult.getStacks()) {
+            for (Output output : st.getOutputs()) {
+                if (output.getOutputKey().equals("PublicDNS")) {
+                    Host host = new Host();
+                    host.setIp(output.getOutputValue());
+                    hosts.add(host);
+                }
+            }
+        }
+        Deployment deployment = new Deployment();
+        deployment.setHosts(hosts);
+        return deployment;
+    }
+
+    /**
+     * This method waits for completion of AWS process and generate result depending on
+     * the result code.
+     *
+     * @param stackBuilder AWS CF builder object.
+     * @param stackName    The stack name to perform the waiting upon.
+     * @return true or false depending on the result.
+     * @throws InterruptedException            when an Interrupt occurs while waiting for CF result.
+     * @throws TestGridInfrastructureException when an error occurs while reading the cf template file.
+     */
+    private boolean waitForAWSProcess(AmazonCloudFormation stackBuilder, String stackName) throws InterruptedException,
+            TestGridInfrastructureException {
+        DescribeStacksRequest wait = new DescribeStacksRequest();
+        wait.setStackName(stackName);
+        boolean completed = false;
+        log.info("Waiting ..");
+        while (!completed) {
+            List<Stack> stacks = stackBuilder.describeStacks(wait).getStacks();
+            if (stacks.isEmpty()) {
+                log.info("There is no stack by the name " + stackName);
+                return false;
+            } else {
+                for (Stack stack : stacks) {
+                    if (stack.getStackStatus().equals(StackStatus.CREATE_COMPLETE.toString()) ||
+                            stack.getStackStatus().equals(StackStatus.DELETE_COMPLETE.toString())) {
+                        log.info("Finished stack task ");
+                        return true;
+                    } else if (stack.getStackStatus().equals(StackStatus.CREATE_FAILED.toString()) ||
+                            stack.getStackStatus().equals(StackStatus.ROLLBACK_FAILED.toString()) ||
+                            stack.getStackStatus().equals(StackStatus.ROLLBACK_COMPLETE.toString())) {
+                        throw new TestGridInfrastructureException("Error while executing infrastructure " +
+                                "command due to :" + stack.getStackStatusReason());
+                    }
+                }
+            }
+            if (!completed) {
```

Comment:
Why do we check the `completed` value here? It looks like the variable value is not changed anywhere. Is there any other way to check whether the stack creation is completed?

Reply:
Changed method.